utilities/transactions/lock/range/range_tree/range_tree_lock_tracker.h

```diff
@@ -0,0 +1,148 @@
+  virtual ~RangeLockList() { clear(); }
+
+  void clear() {
+    for (auto it : buffers_) {
+      it.second->destroy();
+    }
+    buffers_.clear();
+  }
+
+  RangeLockList() : releasing_locks_(false) {}
+
+  void append(uint32_t cf_id, const DBT* left_key, const DBT* right_key) {
+    MutexLock l(&mutex_);
+    // there's only one thread that calls this function.
+    // the same thread will do lock release.
+    assert(!releasing_locks_);
+    auto it = buffers_.find(cf_id);
+    if (it == buffers_.end()) {
+      // create a new one
+      it = buffers_
+               .emplace(cf_id, std::shared_ptr<toku::range_buffer>(
+                                   new toku::range_buffer()))
+               .first;
+      it->second->create();
+    }
+    it->second->append(left_key, right_key);
+  }
+
+  std::unordered_map<uint32_t, std::shared_ptr<toku::range_buffer>> buffers_;
+
+  // Synchronization. See RangeTreeLockManager::UnLock for details
+  port::Mutex mutex_;
+  bool releasing_locks_;
+};
+
+// Tracks range locks
+
+class RangeTreeLockTracker : public LockTracker {
+ public:
+  RangeTreeLockTracker() : range_list(nullptr) {}
+
+  RangeTreeLockTracker(const RangeTreeLockTracker&) = delete;
+  RangeTreeLockTracker& operator=(const RangeTreeLockTracker&) = delete;
+
+  void Track(const PointLockRequest&) override;
+  void Track(const RangeLockRequest&) override;
+
+  bool IsPointLockSupported() const override {
+    // This indicates that we don't implement GetPointLockStatus()
+    return false;
+  }
+  bool IsRangeLockSupported() const override { return true; }
+
+  // a Not-supported dummy implementation.
+  UntrackStatus Untrack(const RangeLockRequest& /*lock_request*/) override {
+    return UntrackStatus::NOT_TRACKED;
+  }
+
+  UntrackStatus Untrack(const PointLockRequest& /*lock_request*/) override {
+    return UntrackStatus::NOT_TRACKED;
+  }
+
+  // "If this method is not supported, leave it as a no-op."
+  void Merge(const LockTracker&) override {}
+
+  // "If this method is not supported, leave it as a no-op."
+  void Subtract(const LockTracker&) override {}
+
+  void Clear() override;
+
+  // "If this method is not supported, returns nullptr."
+  virtual LockTracker* GetTrackedLocksSinceSavePoint(
+      const LockTracker&) const override {
+    return nullptr;
+  }
+
+  PointLockStatus GetPointLockStatus(ColumnFamilyId column_family_id,
+                                     const std::string& key) const override;
+
+  // The return value is only used for tests
+  uint64_t GetNumPointLocks() const override { return 0; }
+
+  ColumnFamilyIterator* GetColumnFamilyIterator() const override {
+    return nullptr;
+  }
+
+  KeyIterator* GetKeyIterator(
+      ColumnFamilyId /*column_family_id*/) const override {
+    return nullptr;
+  }
+
+  // Non-override (TODO: make private!)
+  RangeLockList* getList() { return range_list.get(); }
+  RangeLockList* getOrCreateList();
+
+ private:
+  std::shared_ptr<RangeLockList> range_list;
```

Comment:
nit: `range_list_` to keep style consistent?

Reply:
also, `range_list` is only owned by `LockTracker`? If so, should we make it `unique_ptr`?